### ISSUE TITLE
Suggestiv

### ISSUE THREAD

**Issue by MSXCelulares** [andy-man/ps4-wee-tools#16]
https://github.com/andy-man/ps4-wee-tools/blob/4fc48c8f054bedf43a3a361dfde4a405d8995e47/utils/sflash.py#L415

Hello friend, thank you once again for your excellent work, I was just adapting SPIWAY to offer you this resource but you were faster.
According to the data I have collected so far compared with the data from PSDEVWIKI, I came to this conclusion:
mb_codes = { 2: 'CV', 3: 'SA', 4: 'HV', 5: 'NV', } 4: 'HV' wouldn't be 'HA'?
mb_suffix = All mb_codes 'CV' is 'N', all 'HA' is 'C', 'NV' is 'A', 'B' or 'G' not exist 'NVC'
mb_rev = Revision of board - ??? There is no SAA, SAB, SAC and HAC greater than 001, they are all 001, only SAD, SAE, SAF and NVA, NVB, NVG have revisions above 001.
I'm studying and comparing some DUMPs and I came across this information:
SAF-003 board address 0x1C80D0 contains the following bytes: LVP2C-A0
SAF-004 board address 0x1C80D0 contains the following bytes: LVP3-A0
On 3 SCRAP boards NVG-001 at 0x1C80D0 contain the following bytes: GLC-A0
In dumps from different boards SAA, SAB and SAC boards all 001 at address 0x1C80D0 there is no data - FF FF FF FF FF
Can you check if this is a real thing?
Sorry, but I still don't have SAD, SAE dumps for analysis, but I will continue tracking this mystery.
I would like to suggest this diagram2 file that uses a 3v3 regulator as VCC from NOR SPI, and this small loader executable that comes with the Abkarino Pack to load .hex files into the Teensy.exe
Thank you sir and may God bless you.

![diagram2](https://github.com/andy-man/ps4-wee-tools/assets/96247543/1e54b38f-5471-462e-ac6c-db1aba837f79)
[hw.zip](https://github.com/andy-man/ps4-wee-tools/files/12754246/hw.zip)
[sflash.zip](https://github.com/andy-man/ps4-wee-tools/files/12754247/sflash.zip)

**Comment by andy-man** [comment]
Well good work
**board info** - currently I'm not focused on this task, but thanks for you code.
I'll probably move it to function and add in next commit
**3v3 regulator** - good alternative (will be added to diagram)
**teensy loader** - I don't want to put any exe files (which are not cross platform) into repo
I'll consider adding a link in readme (I mean https://www.pjrc.com/teensy/loader.html)
Perhaps I'll add teensy.exe to release package...
Thanks for your suggestions

**Comment by MSXCelulares** [comment]
Hello friend, I'm sorry I didn't see this before, but I think your Voltage Divider is inverted, blocking GND with resistor R1 could cause reading and writing malfunctions, as the memory IC will not reach a low logic level when necessary . The correct option would be for R1 to be in series with 5VDC. Thank you very much, God bless you.
![diagram](https://github.com/andy-man/ps4-wee-tools/assets/96247543/a073c179-c908-4f76-84f0-17f65af093c7)

**Comment by andy-man** [comment]
Yes you may be right. Voltage divider is not as good as voltage regulator. But according to BwE's scheme it may work. However I didn't test it by myself, I only have my own ps4 10xx and I use pin lift method to r/w onboard.

**Comment by MSXCelulares** [comment]
Hello my friend, I prepared this asset to incorporate into your tool, the Spiway uses a Teensy++ 2.0 and I took the opportunity to make the same teensy become a MonitorSerial, I don't know how to make diagrams as beautiful as yours but I managed to create these, I will do the portability of the syscon_reader code to use the Teensy++ 2.0 instead of an Arduino nano, so your tool will only use one hardware, it will do everything through a Teensy++ 2.0, what do you think this? Big hug, stay with God.
[assets.zip](https://github.com/andy-man/ps4-wee-tools/files/12807637/assets.zip)

**Comment by MSXCelulares** [comment]
Hello Sir, I am finalizing the portability of the Abkarino method to work with PS4_Wee_Tool and Teensy ++ 2.0, the full reading (Fulldump) 512Kb and the NVS_SNVS 128kb reading are now ready, i used a 150r1% resistor its worked fine for me, as soon as possible I will finalize the other functions.
[Abkarino-ps3-wee-files.zip](https://github.com/andy-man/ps4-wee-tools/files/12847286/Abkarino-ps3-wee-files.zip)

**Comment by andy-man** [comment]
I've already integrated Abkarino's syscon flasher in v0.9.0. Read/write/erase/verify just like SPIway screen. Currently I'm testing it and preparing diagrams. There will be a bunch of new small but useful features also. Anyway I'll mention you in credits section for all your efforts!

**Comment by MSXCelulares** [comment]
Thanks Sr. so much.

**Comment by MSXCelulares** [comment]
Hello my friend, excellent work, I suggest changing the syscon Erase function to only erase from block 4 to 511, never erase blocks 0,1,2,3 which are BOOT0, as the chip will lose communication and become inert , dead, I was doing tests here on a separate chip and decided to test the Erase function, it really works perfectly but the chip was dead. I hope I helped, God bless you.
![bad_chip](https://github.com/andy-man/ps4-wee-tools/assets/96247543/9cb2ea5b-95da-45d3-b1f1-7bee0cefe807)

**Comment by H3S4MKH** [comment]
How to run the app?

**Comment by andy-man** [comment]
[Install python](https://www.python.org/downloads/) 3.8 or newer

Install libs _(open terminal i.e. cmd.exe in windows and input these commands)_

`pip install pycryptodome`
`pip install pyserial`

You can also use compiled versions _(see [releases](https://github.com/andy-man/ps4-wee-tools/releases) page)_
All PRO and SE versions comes as EXE file, nothing to install just run.
You can downlaod [v0.9.5](https://github.com/andy-man/ps4-wee-tools/releases/download/v0.9.5/ps4-wee-tools-v0.9.5.zip) to give it a try.

**Comment by kadom11** [comment]
My friend why teensy4 is not supported on your software

**Comment by andy-man** [comment]
Syscon Flasher has teensy 4 support
SPIway for teensy 4 - [this project](https://github.com/alex-joens/TeensySPIFlasher)

I don't have teensy 4 boards to test it
